### DISCUSSION TITLE
How to access resolvers metadata

### DISCUSSION THREAD

**Discussion by mdesousa** [MichalLytek/type-graphql#960] [Q&A]
Hi, I am looking for a good way to query the metadata for the resolvers. For each resolver function I need to get its name and type (query or mutation). Is there an easy way to do this? Thanks!

**Comment by MichalLytek** [comment]
`@Info() info: GraphQLResolveInfo` in resolvers, `{ info }` in middlewares.

**Reply by mdesousa** [reply]
@MichalLytek , sorry don't know if I explained what I'm trying to do correct... i'm trying to read the metadata.
So for example, for the `RecipeResolver` example used in the "Getting started" page I'm trying this code:
```
console.log('metadata', Reflect.getMetadataKeys(RecipeResolver));
const paramTypes = Reflect.getMetadata('design:paramtypes', RecipeResolver);
console.log('paramTypes', paramTypes);
console.log('metadata', Reflect.getMetadataKeys(paramTypes));
```
I was expecting to see a few metadata keys in the RecipeResolver... but all I see is `design:paramtypes`. when I query that it has a function with no metadata. how can I get a list of the `RecipeResolver` resolver functions and their names and types.

Thanks,

**Reply by MichalLytek** [reply]
TypeGraphQL does not emit such metadata, it's Typescript compiler.

The name and type you can get from GraphQL resolve info.

**Reply by glitchgirl** [reply]
Do you have a more complete example of how to use info in the resolvers?
